### PR TITLE
fix: Hotfix-bookings list breaks on unsupported timezone

### LINKED ISSUE #10189: [CAL-2180] Booking with invalid/unsupported timezones goes through
### Issue Summary

Currently, it is possible for a booking to go through with an unsupported Timezone as we can define unsupported timezone in the Booking API POST request in Timezone, and it goes through. Later, when one attempts to open the [cal.com/bookings](http://cal.com/bookings) page, it would hard crash.

### Steps to Reproduce

1. Create a booking using the API with an unsupported Timezone `eg: America/Pacific`. Note that this isn't listed in the UI, but one can create a booking with this timezone via API
2. Once the booking goes through, open `cal.com/bookings` and that should return in hard crash

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Actual Results

* The booking goes through even with unsupported Timezone and later crashes in UI

### Expected Results

Ideally,

* we want to be able to parse and confirm that the timezone passed is supported. If not, we should return error and the booking shouldn't go through.
* We should also handle such situations in the UI a bit better, instead of hard crashing. Also, ideally, we should provide a nice mapping somewhere which allows the user to find out what their timezone is in our DB, like the closest supported alternative

### Technical details

* In production

### Evidence

Reported by a couple of users, confirmed that we **do not** currently have ways to handle it in place.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2180](https://linear.app/calcom/issue/CAL-2180/booking-with-invalidunsupported-timezones-goes-through)</sub>

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

This PR adds a fallback to unsupported attendee timezone which can be fed via the API, resulting in hard crash of the `/bookings` page in the cal.com UI. Currently we can create a booking with certain unsupported timezones via the API, however that is being sorted as well. The code changes in this PR would still be important for backwards compatibility, otherwise `/bookings/past` would remain broken due to old bookings of this nature.

Partly Fixes #10189 

[Loom Video](https://www.loom.com/share/41d770d44fb147978412398627af6a39?sid=8a742664-8f44-4df5-a26b-41c699dbf7ba)
[Loom Video #2](https://www.loom.com/share/ee9210314ef44887ae65daf0f017212d?sid=f3d854c8-5306-478a-9560-a40b613e8090)

## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)
  - [ ] Chore (refactoring code, technical debt, workflow improvements)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] This change requires a documentation update

## How should this be tested?

  - What are the minimal test data to have?
    Create a new booking (either via API or via UI) and make sure that it has a unsupported timezone such as "America/Pacific". You can change the timezone of the attendee by going into the attendee table in prisma studio, if you create the booking via UI. 

  - What is expected (happy path) to have (input and output)?
     The /bookings/ page containing this booking should crash. Once my changes are applied, it shouldn't crash anymore

## Mandatory Tasks

  - [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/lib/date-fns/index.ts [hunk 1]
```diff
@@ -30,6 +30,19 @@ export const formatTime = (
...
+ * @param {string} timeZone - The timezone string to be verified.
+ * @returns {boolean} - Returns 'true' if the provided timezone string is recognized as a valid timezone by the Intl API. Otherwise, returns 'false'.
+ * 
+ */
+export const isSupportedTimeZone = (timeZone: string) => {
+  const validTimezones = Intl.supportedValuesOf("timeZone");
+  return validTimezones.includes(timeZone);
+};
+
+
```
Comment (alishaz-polymath) [inline packages/lib/date-fns/index.ts:45]: Self explanatory

File: packages/ui/components/popover/MeetingTimeInTimezones.tsx [hunk 2]
```diff
@@ -34,8 +35,12 @@ const MeetingTimeInTimezones = ({
...
 }: MeetingTimeInTimezonesProps) => {
   if (!userTimezone || !attendees.length) return null;
-
-  const attendeeTimezones = attendees.map((attendee) => attendee.timeZone);
+  
+  // If attendeeTimezone is unsupported, we fallback to host timezone. Unsupported Attendee timezone can be used due to bad API booking request in the past | backward-compatibility
+  
+  const attendeeTimezones = attendees.map((attendee) => {
+    return isSupportedTimeZone(attendee.timeZone) ? attendee.timeZone : userTimezone;
+  });
```
Comment (alishaz-polymath) [inline packages/ui/components/popover/MeetingTimeInTimezones.tsx:43]: When an unsupported timezone is picked for attendee, we use userTimezone (which is host's timezone) instead.

**Review by alishaz-polymath — COMMENTED** [review]
Self Review added

**Review by emrysal — DISMISSED** [review]

**Review by PeerRich — DISMISSED** [review]

**Review by hariombalhara — APPROVED** [review]
LGTM!! 
There might be other occurences where we might need to fix this but if it solves the immediate problem, it's good to go

**Pull request merged**